model/clusters_mgmt/v1/private_link_cluster_configuration_type.model

```diff
@@ -0,0 +1,23 @@
+/*
+Copyright (c) 2023 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Manages the configuration for the Private Links.
+struct PrivateLinkClusterConfiguration {
+	  // List of additional principals for the Private Link
+    Principals []PrivateLinkPrincipal
```

Comment:
Why aren't you using `PrivateLinkPrincipals`?

Reply:
because that would mean that to use it we need to do 
`privateLinkConfiguration.Principals().Principals().Items()`
with this we can remove a useless level and do just 
`privateLinkConfiguration.Principals().Items()`
which is nicer